### PR TITLE
Beyza  GOK API Week1

### PR DESCRIPTION
**Pull request by beyzagoknl**

### CODE REVIEW

File: 3-UsingAPIs/Week1/homework/ex3-rollDie.js [hunk 1]
```diff
@@ -11,41 +11,33 @@ Full description at: https://github.com/HackYourFuture/Homework/tree/main/3-Usin
...
+      if (roll < randomRollsToDo) {
+        setTimeout(() => rollOnce(roll + 1), 500);
+      }
+    };
+
+    rollOnce(1);
+  });
 }
 
 function main() {
```
Comment (ObadaElSharbatly) [inline 3-UsingAPIs/Week1/homework/ex3-rollDie.js:40]: I think you forgot to refactor this `main` function to deal with the promise coming from the `rollDie` function.
you need to modify this 😀

File: 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js [hunk 2]
```diff
@@ -27,9 +27,8 @@ exercise file.
 const rollDie = require('../../helpers/pokerDiceRoller');
 
 function rollDice() {
-  // TODO Refactor this function
   const dice = [1, 2, 3, 4, 5];
-  return rollDie(1);
+  return Promise.all(dice.map(rollDie));
```
Comment (ObadaElSharbatly) [inline 3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js:31]: prefect

File: 3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js [hunk 3]
```diff
@@ -10,19 +10,28 @@ To throw the dice sequentially we will be using a _promise chain_. Your job is
...
+      return rollDie(3);
+    })
+    .then((value) => {
+      results.push(value);
+      return rollDie(4);
+    })
+    .then((value) => {
+      results.push(value);
+      return rollDie(5);
+    })
```
Comment (ObadaElSharbatly) [inline 3-UsingAPIs/Week1/homework/ex5-pokerDiceChain.js:34]: greate

**Review by ObadaElSharbatly — CHANGES_REQUESTED** [review]
Good work Beyza I think you just need to modify a small thing. but it seems that you did understand the idea of Promises so that's great job 💥💥

**Review by ObadaElSharbatly — APPROVED** [review]

**Comment by beyzagoknl** [comment]
> Good work Beyza I think you just need to modify a small thing. but it seems that you did understand the idea of Promises so that's great job 💥💥

Thank you for your time and your comments.

**Pull request closed without merging**